src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -49,11 +53,62 @@ async function assume_role(req) {
 }
 
 // create and return the signed token
+/**
+ * @param {Object} auth_options
+ * @param {Number} expiry in seconds
+ * @returns {String}
+*/
 function generate_session_token(auth_options, expiry) {
     dbg.log1('sts_post_assume_role.make_session_token: ', auth_options, expiry);
     return jwt_utils.make_auth_token(auth_options, { expiresIn: expiry });
 }
 
+// TODO: Generalize and move to a utils file in the future
+/**
+ * _sts_duration_seconds_checks whether durationSeconds is within AWS' allowed range
+ * @param {Number} input_value
+ */
+function _sts_duration_seconds_check(input_value) {
+    const min_value = 900;
+    const max_value = 43200;
+    let should_throw = false;
+    let constraint = '';
+    let constraint_value = 0;
+    if (input_value < min_value) {
+        should_throw = true;
+        constraint = 'greater';
+        constraint_value = min_value;
+    } else if (input_value > max_value) {
+        should_throw = true;
+        constraint = 'less';
+        constraint_value = max_value;
+    }
+    if (should_throw) {
+        const message_with_details = `Value ${input_value} for durationSeconds failed to satisfy constraint:
+            Member must have value ${constraint} than or equal to ${constraint_value}`;
+        const { code, http_code } = StsError.ValidationError;
+        throw new StsError({ code, message: message_with_details, http_code });
+    }
+}
```

Comment:
I suggest to separate to a parse function like we do for other api's.
Also should have the min/max as configs too.
Also notice to check that this is even parseable as a valid number or else error.
Also no need for moment for some date calculations.

```js
/**
 * @param {string|undefined} duration_input
 * @returns {number} duration in milliseconds
 */
function parse_sts_duration(duration_input) {
    if (duration_input === undefined) {
        return config.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS;
    }

    const duration_sec = Number(duration_input);
    if (!Number.isInteger(duration_sec)) {
        throw new StsError( ?? );  // what error when input is not a number? ValidationError?
    }
    if (duration_sec < config.STS_MIN_DURATION_SECONDS) {
        throw new StsError(sts_duration_validation_error(duration_input, 'greater', config.STS_MIN_DURATION_SECONDS));
    }
    if (duration_sec > config.STS_MAX_DURATION_SECONDS) {
        throw new StsError(sts_duration_validation_error(duration_input, 'less', config.STS_MAX_DURATION_SECONDS));
    }

    const duration_ms = duration_sec * 1000;
    return duration_ms;
}

function sts_duration_validation_error(duration_input, constraint, constraint_value) {
    return {
        ... StsError.ValidationError,
        message: `Value ${duration_input} for durationSeconds failed to satisfy constraint: Member must have value ${constraint} than or equal to ${constraint_value}`,
    };
}

function get_expiration_date(duration_ms) {
    return new Date(Date.now() + duration_ms);
}
```

Reply:
@guymguym why do you set the constants (`STS_MIN_DURATION_SECONDS`, `STS_MAX_DURATION_SECONDS`, `STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS`) in the config file?
and not a file with other constants because those values should not be configurable.